tenant/store.go

```diff
@@ -49,3 +48,149 @@ type Segment struct {
+	tenant.Seq = uint64(len(tenants) + 1)
+	tenantStore.DbStore.Db.Create(tenant)
+	tenantStore.DbStore.Db.NewRecord(*tenant)
+
+	// TODO better way of getting sequence
+	//
+	//	var tenantSeq uint64
+	//	for i := range tenants {
+	//		if tenants[i].Id == myId {
+	//			tenantSeq = uint64(i)
+	//			tenant.Seq = tenantSeq
+	//			tenantStore.DB().Save(tenant)
+	//			log.Printf("Sequence for tenant %s is %d\n", tenants[i].Name, tenantSeq)
+	//			break
+	//		}
+	//	}
+
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (tenantStore *tenantStore) findTenant(id uint64) (Tenant, error) {
+	var tenants []Tenant
+	log.Println("In findTenant()")
+	tenantStore.DbStore.Db.Find(&tenants)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return Tenant{}, err
+	}
+	for i := range tenants {
+		if tenants[i].Id == id {
+			return tenants[i], nil
+		}
+	}
+	return Tenant{}, errors.New("Not found")
+	//	tenant := Tenant{}
+	//	tenantStore.DB().Where("id = ?", id).First(&tenant)
+	//	err := common.MakeMultiError(tenantStore.DB().GetErrors())
+	//	if err != nil {
+	//		return tenant, err
+	//	}
+	//	return tenant, nil
+}
+
+func (tenantStore *tenantStore) addSegment(tenantId uint64, segment *Segment) error {
+	var err error
+
+	// TODO better way of getting sequence
+	var segments []Segment
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	segment.Seq = uint64(len(segments) + 1)
+	tenantStore.DbStore.Db.NewRecord(*segment)
+	segment.TenantId = tenantId
+	tenantStore.DbStore.Db.Create(segment)
+	//	myId := segment.Id
+
+	//	var segmentSeq uint64
+	//
+	//	for i := range segments {
+	//		if segments[i].Id == myId {
+	//			segmentSeq = uint64(i)
+	//			segments[i].Seq = segmentSeq
+	//			log.Printf("Sequence for segment %s is %d\n", segments[i].Name, segmentSeq)
+	//			tenantStore.DB().Save(segment)
+	//			break
+	//		}
+	//	}
+
+	err = common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (tenantStore tenantStore) CreateSchemaPostProcess() error {
+	return nil
+}
+
+func (tenantStore *tenantStore) findSegment(tenantId uint64, id uint64) (Segment, error) {
+	var segments []Segment
+	log.Println("In listSegments()")
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return Segment{}, err
+	}
+	for i := range segments {
+		if segments[i].Id == id {
+			return segments[i], nil
+		}
+	}
+	return Segment{}, errors.New("Not found")
+	//	segment := Segment{}
+	//	tenantStore.DB().Where("tenant_id = ? AND id = ?", tenantId, id).First(&segment)
+	//	err := common.MakeMultiError(tenantStore.DB().GetErrors())
+	//	if err != nil {
+	//		return segment, err
+	//	}
+	//	return segment, nil
+}
+
+//func (mysqlStore *mysqlStore) listSegments() ([]Tenant, error) {
```

Comment:
There's quite a bit code around here that's been commented out. Can we remove it?